client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -65,6 +73,11 @@ private final class PoolManager[F[_], A <: Connection[F]](
   private def numConnectionsCheckHolds(key: RequestKey): Boolean =
     curTotal < maxTotal && allocated.getOrElse(key, 0) < maxConnectionsPerRequestKey(key)
 
+  private def isExpired(k: RequestKey, t: Instant): Boolean =
+    waitExpiryTime(k) != -1 && t
+      .plus(waitExpiryTime(k).toLong, ChronoUnit.SECONDS)
+      .isBefore(Instant.now())
```

Comment:
This gets a lot cleaner if the duration suggestion is followed.  The `-1` is `!_.isFinite`, and there's a clear time unit.

Reply:
Ok.